pandas/tests/groupby/test_groupby.py

```diff
@@ -2059,6 +2060,93 @@ def test_groups_repr_truncates(max_seq_items, expected):
         assert result == expected
 
 
+@td.skip_if_no("lxml")
+def test_groupby_repr():
+    """
+    All groups and all rows in a group are shown in html output.
+    """
+    n_groups = 5
+    length = n_groups * 5
+    df = DataFrame(
+        {
+            "A": range(length),
+            "B": range(0, length * 2, 2),
+            "C": list(range(n_groups)) * (length // n_groups),
+        }
+    )
+
+    df_groupby = df.groupby("C")
+    html_groupby = df_groupby._repr_html_()
+
+    dfs_from_html = pd.read_html(StringIO(html_groupby), index_col=0)
+
+    for k, (group_name, df_group) in enumerate(df_groupby):
+        tm.assert_frame_equal(dfs_from_html[k], df_group)
+
+
+@td.skip_if_no("lxml")
+def test_groupby_repr_truncated_group():
+    """
+    In the groups not all rows are shown in html output.
+    """
+    n_groups = 10
+    length = n_groups * 20
+
+    df = DataFrame(
+        {
+            "A": range(length),
+            "B": range(0, length * 2, 2),
+            "C": list(range(n_groups)) * (length // n_groups),
+        }
+    )
+
+    df_groupby = df.groupby("C")
+    html_groupby = df_groupby._repr_html_()
+
+    dfs_from_html = pd.read_html(StringIO(html_groupby), index_col=0)
+
+    # For each group only test first and last row
+    for k, (group_name, df_group) in enumerate(df_groupby):
+        dtype = df_group.iloc[0].dtype
+        tm.assert_series_equal(
+            dfs_from_html[k].iloc[0].astype(dtype), df_group.iloc[0], check_names=False
+        )
+        tm.assert_series_equal(
+            dfs_from_html[k].iloc[-1].astype(dtype),
+            df_group.iloc[-1],
+            check_names=False,
+        )
+
+
+@td.skip_if_no("lxml")
+def test_groupby_repr_not_all_groups():
+    """
+    Not all groups are shown in html output.
+    """
+    n_groups = 30
+    length = n_groups * 5
+    df = DataFrame(
+        {
+            "A": range(length),
+            "B": range(0, length * 2, 2),
+            "C": list(range(n_groups)) * (length // n_groups),
+        }
+    )
+
+    df_groupby = df.groupby("C")
+    html_groupby = df_groupby._repr_html_()
+
+    dfs_from_html = pd.read_html(StringIO(html_groupby), index_col=0)
+
+    # Test first and last group
```

Comment:
Same as above.

Reply:
The comment is almost exactly the same as in test_groupby_repr_truncated_group. Should I rather refer to that comment?